modules/develop/pages/data-transforms/how-transforms-work.adoc

```diff
@@ -0,0 +1,41 @@
+= How Data Transforms Work
+:description: Learn how Redpanda data transforms work.
+
+Redpanda provides the framework to build and deploy inline transformations on data written to Redpanda topics, delivering processed and validated data to consumers in the format they expect. Redpanda does this directly inside the broker, eliminating the need to manage a separate stream processing environment or use 3rd-party tools. 
+
+image::shared:wasm1.png[Data transforms in a broker] 
+
+To learn how to build and deploy data transforms, see xref:./run-transforms.adoc[].
+
+== Data transforms with WebAssembly
+
+Data transforms use WebAssembly (Wasm) engines inside a Redpanda broker, allowing Redpanda to control the entire transform lifecycle. For example, Redpanda can stop and start transforms when partitions are moved or to free up system resources for other tasks. 
+
+Data transforms take an input topic and map it to an output topic, running a virtual machine (VM) on the same shard as every partition leader for a topic. Redpanda uses just-in-time (JIT) compilation, which compiles the function in memory, writes it to executable space, then runs the directly translated machine code. 
+
+When you deploy a data transform, Redpanda pushes the generated `.wasm` files and metadata to the cluster, which then stores and replicates the bytecode and metadata using internal broker topics. Each shard of Redpanda runs its own copy of the transform function. There is at most one runtime for a transform for any shard, and memory for each function is reserved within the broker with the `wasm_per_core_memory_reservation` and `wasm_per_function_memory_limit` properties. CPU time is dynamically allocated to the Wasm runtime and ensures that the code does not run forever and cannot block the broker from handling traffic or doing other work, such as Tiered Storage uploads. 
```

Comment:
I was looking for details about metadata here. So took the same example from later in this doc. Also, split the resource management items out into bullets to make it easier to scan.

Reply:
The style guide also says to avoid referring to file types like `.wasm`, and I don't think it's necessary to refer to the files anyway 🤷 as the important part is the bytecode.